ext/msgpack/packer.c

```diff
@@ -121,19 +121,56 @@ void msgpack_packer_write_hash_value(msgpack_packer_t* pk, VALUE v)
 #endif
 }
 
+struct msgpack_call_proc_args_t;
+typedef struct msgpack_call_proc_args_t msgpack_call_proc_args_t;
```

Comment:
Does it have any effect than defining a usual struct `msgpack_call_proc_args_t`?

Reply:
I don't think so no, it's just a common way to pass more than one argument through `rb_protect`, you can see this pattern a bit everywhere in MRI and most extensions.